proparse/src/main/java/org/prorefactor/treeparser/TreeParserRootSymbolScope.java

```diff
@@ -139,6 +141,34 @@ public TableBuffer getLocalTableBuffer(ITable table) {
     return bufferMap.get(table.getName().toLowerCase());
   }
 
+  @Override
+  public Variable lookupVariable(String name) {
+    Variable var = super.lookupVariable(name);
+    if (var != null) {
+      return var;
+    }
+
+    TypeInfo info = typeInfo;
+    while (info != null) {
+      if (info.hasProperty(name)) {
+        return new Variable(name, this);
+      }
+      info = refSession.getTypeInfo(info.getParentTypeName());
+    }
+    return null;
+  }
+
+  @Override
+  public Dataset lookupDataset(String name) {
+    Dataset ds = super.lookupDataset(name);
+    if (ds != null) {
+      return ds;
+    }
+    
+    // TODO Lookup in parent classes
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1135)